tracee-ebpf/tracee/tracee.bpf.c

```diff
@@ -1999,6 +2000,66 @@ struct bpf_raw_tracepoint_args *ctx
     return 0;
 }
 
+SEC("raw_tracepoint/sched_process_exec")
+int tracepoint__sched__sched_process_exec(
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4, 17, 0)
+void *ctx
+#else
+struct bpf_raw_tracepoint_args *ctx
+#endif
+)
+{
+    if (!should_trace())
+        return 0;
+
+    u32 pid = bpf_get_current_pid_tgid();
+    // Remove pid from traced_pids_map
+    bpf_map_delete_elem(&traced_pids_map, &pid);
+
+    if (get_config(CONFIG_NEW_CONT_FILTER)) {
+        struct task_struct *task;
+        task = (struct task_struct *)bpf_get_current_task();
+
+        u32 pid_ns = get_task_pid_ns_id(task);
+        if (get_task_ns_pid(task) == 1) {
+            // If pid equals 1 - stop tracing this pid namespace
+            bpf_map_delete_elem(&new_pidns_map, &pid_ns);
+        }
+    }
+    if (get_config(CONFIG_NEW_PID_FILTER)) {
+        // Remove pid from new_pids_map
+        bpf_map_delete_elem(&new_pids_map, &pid);
+    }
+
+    buf_t *submit_p = get_buf(SUBMIT_BUF_IDX);
+    if (submit_p == NULL)
+        return 0;
+    set_buf_off(SUBMIT_BUF_IDX, sizeof(context_t));
+
+    context_t context = init_and_save_context(ctx, submit_p, SCHED_PROCESS_EXEC, 2, 0);
+    struct task_struct *p = (struct task_struct *)ctx->args[0];
+    struct linux_binprm *bprm = (struct linux_binprm *)ctx->args[2];
+
+    if (bprm == NULL) {
+        return -1;
+    }
+    struct task_struct *parent = (struct task_struct *)READ_KERN(p->parent);
+    if (parent == NULL) {
+            return -1;
+    }
+    const char *filename = (const char *)READ_KERN(bprm->filename);
```

Comment:
Reading values of internal kernel structures may change between different kernel versions.
You should:
1. Move these to the helper functions section
2. Check that the values you read are located and named the same in different kernel versions (starting from 4.18 to the latest version). If not, add a compilation flag according to their location for a specific kernel version